src/Dialog/Dialog.tsx

```diff
@@ -310,61 +298,211 @@ const _Dialog = React.forwardRef<HTMLDivElement, React.PropsWithChildren<DialogP
+          height={height}
+          ref={dialogRef}
+          role={role}
+          aria-labelledby={dialogLabelId}
+          aria-describedby={dialogDescriptionId}
+          aria-modal
+          sx={sx}
+        >
+          {header}
+          <ScrollableRegion aria-labelledby={dialogLabelId} className="DialogOverflowWrapper">
+            {body}
+          </ScrollableRegion>
+          {footer}
+        </NormalDialog>
+      </Backdrop>
+    </Portal>
   )
 })
 _Dialog.displayName = 'Dialog'
 
+const Buttons: React.FC<React.PropsWithChildren<{buttons: DialogButtonProps[]}>> = ({buttons}) => {
+  const autoFocusRef = useProvidedRefOrCreate<HTMLButtonElement>(buttons.find(button => button.autoFocus)?.ref)
+  let autoFocusCount = 0
+  const [hasRendered, setHasRendered] = useState(0)
+  useEffect(() => {
+    // hack to work around dialogs originating from other focus traps.
+    if (hasRendered === 1) {
+      autoFocusRef.current?.focus()
+    } else {
+      setHasRendered(hasRendered + 1)
+    }
+  }, [autoFocusRef, hasRendered])
+
+  return (
+    <>
+      {buttons.map((dialogButtonProps, index) => {
+        const {content, buttonType = 'default', autoFocus = false, ...buttonProps} = dialogButtonProps
+        return (
+          <Button
+            key={index}
+            {...buttonProps}
+            // 'normal' value is equivalent to 'default', this is used for backwards compatibility
+            variant={buttonType === 'normal' ? 'default' : buttonType}
+            ref={autoFocus && autoFocusCount === 0 ? (autoFocusCount++, autoFocusRef) : null}
+          >
+            {content}
+          </Button>
+        )
+      })}
+    </>
+  )
+}
+
+/**
+ * Styled wrappers
+ */
+
+const Backdrop = styled('div')`
+  position: fixed;
+  top: 0;
+  left: 0;
+  bottom: 0;
+  right: 0;
+  display: flex;
+  align-items: center;
+  justify-content: center;
+  background-color: ${get('colors.primer.canvas.backdrop')};
+  animation: dialog-backdrop-appear ${ANIMATION_DURATION} ${get('animation.easeOutCubic')};
+
+  @keyframes dialog-backdrop-appear {
+    0% {
+      opacity: 0;
+    }
+    100% {
+      opacity: 1;
+    }
+  }
+`
+
+const NormalDialog = styled.div<StyledDialogProps>`
+  display: flex;
+  flex-direction: column;
+  background-color: ${get('colors.canvas.overlay')};
+  box-shadow: ${get('shadows.overlay.shadow')};
+  min-width: 296px;
+  max-width: calc(100vw - 64px);
+  max-height: calc(100vh - 64px);
+  width: ${props => widthMap[props.width ?? ('xlarge' as const)]};
+  height: ${props => heightMap[props.height ?? ('auto' as const)]};
+  border-radius: 12px;
+  opacity: 1;
+  animation: overlay--dialog-appear ${ANIMATION_DURATION} ${get('animation.easeOutCubic')};
+
+  @keyframes overlay--dialog-appear {
+    0% {
+      opacity: 0;
+      transform: scale(0.5);
+    }
+    100% {
+      opacity: 1;
+      transform: scale(1);
+    }
+  }
+
+  ${sx};
+`
+
+const FullScreenDialog = styled.div<StyledDialogProps>`
```

Comment:
Is this meant to be a `dialog` instead of a `div`?

Reply:
If these _are_ meant to be `dialog`s, we'll need to update the ref types to `HTMLDialogComponent` like I suggested in `src/Dialog/DialogBottomSheet.tsx`

Reply:
Actually I had to revert this again because the role is defined through the `role` prop.